README.md

```diff
@@ -1349,6 +1349,31 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
 * <a id='newline-at-eof'></a>(<a href='#newline-at-eof'>link</a>) **Files should end in a newline.** [![SwiftLint: trailing_newline](https://img.shields.io/badge/SwiftLint-trailing__newline-007A87.svg)](https://github.com/realm/SwiftLint/blob/master/Rules.md#trailing-newline)
 
+* <a id='mark-types-and-extensions'></a>(<a href='#mark-types-and-extensions'>link</a>) **Each type and extension should be preceded by a `MARK` comment.** [![SwiftFormat: markTypes](https://img.shields.io/badge/SwiftFormat-markTypes-008489.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#markTypes)
+  * Types should be preceded by a `// MARK: - TypeName` comment.
+  * Extensions that add a conformance should be preceded by a `// MARK: TypeName + ProtocolName` comment.
+  * Extensions that follow the type being extended should omit that type's name and instead use `// MARK: ProtocolName`.
+  * If there is only one type or extension in a file, the `MARK` comment can be omitted.
+  * For extensions that do not add new conformances, consider adding a `// MARK:` with a descriptive comment.
+
+  <details>
+
+  ```swift
+  // MARK: - GalaxyView
+
+  final class GalaxyView: UIView { }
+
+  // MARK: ContentConfigurableView
+
+  extension GalaxyView: ContentConfigurableView { }
+
+  // MARK: Galaxy + SpaceThing
+
+  extension Galaxy: SpaceThing { }
```

Comment:
What happens for `extension Galaxy: SpaceThing, ContainsDarkMatter { }`?

Reply:
The [SwiftFormat rule](https://github.com/nicklockwood/SwiftFormat/pull/747/files#diff-68bc6d56aad81543185711914b2a1424R1675) uses a comma-separated list of the conformances:

```diff
+ // MARK: Galaxy + SpaceThing, DarkMatterContaining
+
 extension Galaxy: SpaceThing, DarkMatterContaining { }
```

Reply:
If we're adding commas between protocols it feels like we should just have the mark contents match the Swift declaration since folks are familiar with that syntax, e.g. `Galaxy: SpaceThing, DarkMatterContaining`. Using a plus feels less precise, especially since we're only using it for protocol conformances and not other types of marks

Reply:
Good suggestion, that sounds reasonable. I can update the SwiftFormat rule to support this level of customization.

Reply:
although this does mean that extension marks would have two colons:

```swift
// MARK: Galaxy: SpaceThing, DarkMatterContainig
```

Reply:
I think we should prefer the `+`. Having two `:` characters feels odd. I think the `+` also is a convention people will understand. In the Objective-C days it was very common to use a `+` in file names when a file implemented a category (similar to a protocol), e.g. `MyObjCCode+Category.h`/`MyObjCCode+Category.m`.

Reply:
I can see the argument for the plus—sounds good to me.